add HCP for Z (Editing/rewriting in process)

krishnanshruthi opened the pull request:
https://issues.redhat.com/browse/ACM-7922 (This is not a RHACM doc issue, seems to be a feature outside of RHACM. I am reaching out for more info)

krishnanshruthi committed:
ACM-7922

krishnanshruthi committed:
ACM-7922

krishnanshruthi committed:
ACM-7922

swopebe commented:
Hi. For info:

We do need all of our PRs to be linked to the **RHACM doc issue** and **dev feature**, especially for new content so that we ensure that doc is only documenting what is prioritized for the release:

Please see that **process** on our playbook:
https://docs.google.com/document/d/1YTqpZRH54Bnn4WJ2nZmjaCoiRtqmrc2w6DdQxe_yLZ8/edit#heading=h.9fvyr2rdriby


This is because we maintain the repo, all bugs, build etc...

This big of a change should have gone come to us in the way of a known feature and been added by a RHACM writer, to help with the editing. Such a large PR will need to be accounted for in our scrums/release work. We work closely with development to maintain that process and work.

Please link the RHACM doc issue here along with the feature. Let's please discuss any future work like this that will be sent to the repo. 

@cadawson FYI.

swopebe commented:
@cadawson I am looking for the _RHACM_ doc issue that should have been opened for us? (per the process of adding new features) 

We need _PM and QE validation_, just as we treat all new features. 

(This late in the release, I would hope we had something like this slated in the train. There's a gap that hopefully Doc Guild can address in the future.) :-)

krishnanshruthi committed:
ACM-7922

swopebe commented:
Discussion for short term:

We all agree that for the contribution from IBM, we do need confirmation and QE resources to say this has been tested, as the issues that come from this doc/feature will land in the RHACM doc/dev bucket. 

Not yet ready for support matrix or What's new, we will hold on this PR until we have more discussion.

Long term: feature process will be discussed in Doc Guild.

krishnanshruthi committed:
readded the removed chunk from earlier commit

swopebe commented:
Note to consider. With this PR, I cannot easily check out to edit in VS Code. It is much easier to edit in VS Code in my opinion, so for large changes, I think it may be better to either not fork or to give us the content. I will edit in the browser for this one.

swopebe committed:
Editing on RHACM side--bcs

coded: `api.${HOSTED_CLUSTER_NAME}.${BASEDOMAIN}''`

Using directional language is discouraged in the style manuals. Removed "above" and changed to "In the previous example..."

We don't use "we" in technical doc, changed that line (27).

ip changed to IP address.

I still have a couple questions that I will use the comment feature to send.

swopebe commented on clusters/hosted_control_planes/config_dns_bare-metal_ibmz.adoc:
```diff
@@ -0,0 +1,29 @@
+The DNS entry can be as simple as a record that points to one of the nodes in the managed cluster that is running the hosted control plane. The entry can also point to a load balancer that is deployed to redirect incoming traffic to the Ingress pods.
+
+See the following example of DNS configuration:
+----
```
Question about the # and @ and $, as we don't typically add those in formal doc.

swopebe replied:
Break up command, output, and files from commands. Remove # cat if we can for formal doc.

swopebe replied:
* .apps in this area will throw of our format in the PR, we try to fix that where we can. Laura/Brandi.

swopebe commented on clusters/hosted_control_planes/config_dns_bare-metal_ibmz.adoc:
```diff
@@ -0,0 +1,29 @@
+                      IN NS bastion.example.krnl.es.com.
+api                   IN A 1xx.2x.2xx.1xx
+api-int               IN A 1xx.2x.2xx.1xx
+*.apps                IN A 1xx.2x.2xx.1xx
```
`*.apps` throws off our PRs and we had a solution, I just have to remember or find what we did for that.

swopebe replied:
@krishnanshruthi we need to break this our as we did here:

https://github.com/stolostron/rhacm-docs/blob/2.9_stage/clusters/hosted_control_planes/create_hosted_clusters_kubevirt_default_ingress_dns.adoc?plain=1

Thanks @lahinson for that info.

swopebe commented on clusters/hosted_control_planes/config_dns_bare-metal_ibmz.adoc:
```diff
@@ -0,0 +1,33 @@
+
+*Note:* In the previous example, `1xx.2x.2xx.1xx` is a load balancer IP address that handles ingress and egress traffic for hosted control plane.
+
+For zvm, add additional IP addresses that correspond to the IP address of the agent.
```
What is zvm? We probably need to write this correctly either as code or a proper title.

swopebe replied:
We also need to properly introduce what we are showing on line 28.

swopebe committed:
Update config_dns_bare-metal_ibmz.adoc

swopebe committed:
Update configure_hosted_aws.adoc

swopebe committed:
Update configure_hosted_aws.adoc

swopebe commented on clusters/hosted_control_planes/configure_hosted_aws.adoc:
```diff
@@ -28,6 +30,7 @@ oc get managedclusters local-cluster
 * The AWS command line interface. 
 
 * The link:https://access.redhat.com/documentation/en-us/red_hat_advanced_cluster_management_for_kubernetes/2.9/html/clusters/cluster_mce_overview#hosted-install-cli[hosted control plane command line interface (hcp) binary].
+//why is this https--bcs (internal link)
```
We need to link from within the doc here.

swopebe committed:
Update link to internal doc, title changed

swopebe commented on clusters/hosted_control_planes/configure_hosted_bm_ibmz.adoc:
```diff
@@ -0,0 +1,71 @@
+[#configuring-hosting-service-cluster-configure-ibmz]
```
Note for me to pick up here with removing note and labeling tech preview as the rest of the doc.

swopebe commented on clusters/hosted_control_planes/configure_hosted_aws.adoc:
```diff
@@ -27,7 +29,7 @@ oc get managedclusters local-cluster
 * The AWS command line interface. 
 
-* The link:https://access.redhat.com/documentation/en-us/red_hat_advanced_cluster_management_for_kubernetes/2.9/html/clusters/cluster_mce_overview#hosted-install-cli[hosted control plane command line interface (hcp) binary].
+* The xref:xref:../clusters/hosted_control_planes/hosted_install_cli.adoc#hosted-install-cli[Installing the hosted control plane command line interface].
```
link changed

lahinson replied:
```suggestion
* The xref:../clusters/hosted_control_planes/hosted_install_cli.adoc#hosted-install-cli[Installing the hosted control plane command line interface].
```

swopebe committed:
numerous changes/

swopebe commented on clusters/hosted_control_planes/configure_hosted_bm_ibmz.adoc:
```diff
@@ -0,0 +1,71 @@
+
+* DNS: The API and Ingress endpoints must be routable.
+
+[#configuring-hcp-bare-metal]
```
we title this configuring, but most of the topic is requirements, then the actual configuring is links to other topics, so let's explore changing this format.

swopebe commented on clusters/hosted_control_planes/hosted-bm-agents-ibmz.adoc:
```diff
@@ -0,0 +1,103 @@
+[#hosted-bare-metal-adding-agents-ibmz]
+= Adding IBM Z agents to the InfraEnv resource (Technology Preview)
+
```
We definitely need a short description here.

swopebe replied:
Command format should include `<proper-style-syntax>`

swopebe commented on clusters/hosted_control_planes/hosted-bm-agents-ibmz.adoc:
```diff
@@ -0,0 +1,103 @@
+[#hosted-bare-metal-adding-agents-ibmz]
+= Adding IBM Z agents to the InfraEnv resource (Technology Preview)
+
+- Boot zKVM and zVM with the downloaded PXE images from the 'InfraEnv' resource. Wait for the Agents to get created on the management cluster. At startup, the host communicates with the Assisted Service and registers as an agent in the same namespace as the 'InfraEnv' resource. 
```
Is this a step? Not sure I understand what is going on in the beginning of this topic?

swopebe replied:
When we give the user a command to run with lots of variables, we need to tell them what to replace. Do we need all of this in the command on line 9?

swopebe commented on clusters/hosted_control_planes/hosted-bm-agents-ibmz.adoc:
```diff
@@ -0,0 +1,103 @@
+
+. Update the paramfile to add 'rootfs url', 'net type' and disk type. 
+
+Following is the sample command for 'vswitch' network with fcp singlepath/ dasd disk:
```
what is meant by _fcp singlepath/ dasd disk_?

SNiemann15 replied:
FCP and DASD are disk types on IBM Z machines. Not sure what is meant with `singlepath`
vswitch is written as follows: VSwitch

swopebe commented on clusters/hosted_control_planes/hosted-bm-agents-ibmz.adoc:
```diff
@@ -0,0 +1,103 @@
++
+----
+installerArgs: '[ "--append-karg", "rd.neednet=1", "--append-karg", "ip=<Guest vm ip>::<nameserver>:255.255.255.0:<hostname>","--append-karg", "nameserver=<nameserver>", "--append-karg", "rd.znet=qeth,<network adaptor range>,layer2=1","--append-karg", "rd.<storage type>=<storage>" ]'
+----
```
I am a bit disoriented in this file and not sure of the process. Let's revisit this and create 1. a solid description that tells the user what they are doing and why, and then clear steps, removing assumptions from abbreviations.

swopebe committed:
changes to the procedure and introducing the commands and example

swopebe commented on clusters/hosted_control_planes/managing_hosted_bm_ibmz.adoc:
```diff
@@ -0,0 +1,16 @@
+[#hosted-control-planes-manage-bm-ibmz]
+= Managing hosted control plane clusters on bare metal (Technology Preview)
```
We need to revisit here bc its a file called _managing_ and it has _prereqs_ and _additional resources_, there is not really a task here to warrant a file...let's see how we can reformat, please?

swopebe committed:
Update managing_hosted_bm_ibmz.adoc

swopebe commented on clusters/hosted_control_planes/managing_hosted_bm_ibmz.adoc:
```diff
@@ -0,0 +1,16 @@
+
+[#additional-resources-manage-bm]
+=== Additional resources
+
```
This section is for additional links, not follow on processes, so let's discuss moving this, though I think this whole file may need to be merged inside an actual task.

lahinson replied:
This seems to be a copy of the `managing_hosted_bm.adoc` file, only without any follow-on tasks. I agree with @swopebe that this file can be removed from this PR.

swopebe reviewed (changes requested):
I am working this and seeing that it needs a lot of change. I am editing where it is clear and leaving comments. Please don't change anything just yet until I get through it. I am just keeping you updated. We probably need a meeting to address a few things. I will organize that when I get a chance.

Neeraj8418 commented on clusters/hosted_control_planes/configure_hosted_aws.adoc:
```diff
@@ -43,6 +43,6 @@ oc get managedclusters local-cluster
 * The hosted control planes feature is enabled by default. To disable the hosted control planes feature or, if you already disabled it and want to manually enable it, see xref:../hosted_control_planes/enable_or_disable_hosted.adoc#enable-or-disable-hosted-control-planes[Enabling or disabling the hosted control planes feature].
 
-* To manage hosted clusters by running {aap} jobs, see xref:../cluster_lifecycle/ansible_config_hosted_cluster.adoc#ansible-config-hosted-cluster[Configuring {aap-short} jobs to run on hosted clusters].
+* You can now deploy the SR-IOV Operator. For more information, see link:https://access.redhat.com/documentation/en-us/openshift_container_platform/4.14/html/networking/hardware-networks#sriov-operator-hosted-control-planes_configuring-sriov-operator[Deploying the SR-IOV Operator for hosted control planes].
```
https://access.redhat.com/documentation/en-us/openshift_container_platform/4.14/html/networking/hardware-networks#sriov-operator-hosted-control-planes_configuring-sriov-operator ==> Is this the correct url?

swopebe replied:
I think so, but they aren't GA, yet. I think today?

Neeraj8418 commented on clusters/hosted_control_planes/hosted-bm-agents-ibmz.adoc:
```diff
@@ -0,0 +1,103 @@
+
++
+----
+vmur pun -r -u m1321006 -N kernel.img $INSTALLERKERNELLOCATION/<image name>
```
vmur pun -r -u <guest machine> -N kernel.img $INSTALLERKERNELLOCATION/<image name>. ==>remove the name of guest machine

Neeraj8418 commented on clusters/hosted_control_planes/hosted-bm-agents-ibmz.adoc:
```diff
@@ -0,0 +1,103 @@
++
+----
+vmur pun -r -u m1321006 -N kernel.img $INSTALLERKERNELLOCATION/<image name>
+vmur pun -r -u m1321006 -N generic.parm $PARMFILELOCATION/paramfilename
```
same as above

Neeraj8418 commented on clusters/hosted_control_planes/hosted-bm-agents-ibmz.adoc:
```diff
@@ -0,0 +1,103 @@
+----
+vmur pun -r -u m1321006 -N kernel.img $INSTALLERKERNELLOCATION/<image name>
+vmur pun -r -u m1321006 -N generic.parm $PARMFILELOCATION/paramfilename
+vmur pun -r -u m1321006 -N initrd.img $INSTALLERINITRAMFSLOCATION/<image name>
```
same as above

Neeraj8418 commented on clusters/hosted_control_planes/hosted-bm-agents-ibmz.adoc:
```diff
@@ -0,0 +1,103 @@
+After each agent is created, you can optionally set the agent ID 'installation_disk_id' and hostname in the specification and approve the agent by entering the following commands:
++
+----
+oc -n ${HOSTED_CONTROL_PLANE_NAMESPACE} patch agent 50c23cda-cedc-9bbd-bcf1-9b3a5c75804d -p '{"spec":{"installation_disk_id":"/dev/sda","approved":true,"hostname":"worker-zvm-0.hostedn.example.com"}}' --type merge
```
oc -n ${HOSTED_CONTROL_PLANE_NAMESPACE} patch agent 50c23cda-cedc-9bbd-bcf1-9b3a5c75804d -p '{"spec":{"installation_disk_id":"/dev/sda","approved":true,"hostname":"worker-zvm-0.example.krnl.es"}}' --type merge. ==> The hostname can be different but change basedomain to krnl.es to avoid confusion

phani2898 commented on clusters/hosted_control_planes/config_dns_bare-metal_ibmz.adoc:
```diff
@@ -0,0 +1,29 @@
+= Configure DNS for hosted control plane
+
+
+The API Server for the hosted cluster is exposed as a 'NodePort' service. A DNS entry must exist for 'api.${HOSTED_CLUSTER_NAME}.${BASEDOMAIN}'' that points to the destination where the API Server can be reached.
```
Extra ' is present in here `'api.${HOSTED_CLUSTER_NAME}.${BASEDOMAIN}''`

phani2898 commented on clusters/hosted_control_planes/config_dns_bare-metal_ibmz.adoc:
```diff
@@ -0,0 +1,29 @@
+In the above example, 1xx.2x.2xx.1xx is a load balancer ip which handles ingress and egress traffic for hosted control plane.
+
+For zvm we have to add additional IP corresponding to the ip of agent.
+worker-0              IN A 1xx.2x.2xx.1yy
```
I think we should replace `worker` with `compute` in every documentation as being followed

swopebe replied:
Worked in the past but may not apply here: 
```
[source,bash]
----
*.apps.mgmt-cluster.example.com
----
```

krishnanshruthi committed:
incorporated vikas's inputs

swopebe commented on clusters/hosted_control_planes/scaling_nodepool_hosted_ibmz.adoc:
```diff
@@ -0,0 +1,124 @@
+[#scaling-the-nodepool-ibmz]
+= Scaling the NodePool object for a hosted cluster on IBM Z
+
+You add nodes to your hosted cluster by scaling the NodePool object.  
```
Needs bit more short description here.

swopebe replied:
Why does the user want to scale.

swopebe commented on clusters/hosted_control_planes/scaling_nodepool_hosted_ibmz.adoc:
```diff
@@ -0,0 +1,124 @@
+
+You add nodes to your hosted cluster by scaling the NodePool object.  
+
+. Scale the NodePool object to two nodes:
```
Run the following command to...

swopebe commented on clusters/hosted_control_planes/scaling_nodepool_hosted_ibmz.adoc:
```diff
@@ -0,0 +1,124 @@
+* `installing-in-progress`
+* `added-to-existing-cluster`
+* 'oc -n ${HOSTED_CONTROL_PLANE_NAMESPACE} get agent'
+----
```
Format issues here

swopebe committed:
Update configure_hosted_bm_ibmz.adoc space added

swopebe commented on clusters/hosted_control_planes/hosted-bm-agents-ibmz.adoc:
```diff
@@ -0,0 +1,103 @@
+
+- Boot KVM on IBM Z and z/VM with the downloaded PXE images from the 'InfraEnv' resource. Wait for the Agents to get created on the management cluster. At startup, the host communicates with the Assisted Service and registers as an agent in the same namespace as the 'InfraEnv' resource. 
+- For KVM on IBM Z, perform the boot by running the following command:
++
```
space here

swopebe replied:
This command I have questions about; needs to be cleaned up and explained.

swopebe commented on clusters/hosted_control_planes/hosted-bm-agents-ibmz.adoc:
```diff
@@ -0,0 +1,103 @@
+
+Following is the sample command for 'ROCE' network with fcp singlepath/ dasdc disk:
++
+----
```
format issue

swopebe commented on clusters/hosted_control_planes/scaling_nodepool_hosted_ibmz.adoc:
```diff
@@ -0,0 +1,124 @@
++
+The Cluster API agent provider randomly picks two agents that are then assigned to the hosted cluster. Those agents go through different states and finally join the hosted cluster as OpenShift Container Platform nodes. The agents pass through states in the following order:
+
++ 
```
following is a list and a command, not sure what is needed in this section

swopebe commented on clusters/hosted_control_planes/scaling_nodepool_hosted_ibmz.adoc:
```diff
@@ -0,0 +1,124 @@
+
+----
+
+Sample output:
```
format issue

swopebe committed:
Update config_dns_bare-metal_ibmz.adoc command change syntax.

swopebe commented on clusters/hosted_control_planes/config_dns_bare-metal_ibmz.adoc:
```diff
@@ -0,0 +1,35 @@
+[#configuring-dns-hosted-control-plane-ibmz]
+= Configure DNS for hosted control plane
+
```
double check command on line 4

swopebe committed:
Update config_dns_bare-metal_ibmz.adoc edits

swopebe commented on clusters/hosted_control_planes/configure_hosted_aws.adoc:
```diff
@@ -5,7 +5,9 @@ To configure hosted control planes, you need a hosting cluster and a hosted clus
 
 The {mce-short} 2.4 supports only the default `local-cluster` and the hub cluster as the hosting cluster.
```
Remove this file in the fork.

swopebe commented on clusters/hosted_control_planes/configure_hosted_bm.adoc:
```diff
@@ -72,4 +72,4 @@ After you meet the prerequisites, complete the following steps to configure host
 
```
remove from PR

swopebe committed:
Update configure_hosted_bm_ibmz.adoc ibmz

krishnanshruthi committed:
as per inputs by Brandi

SNiemann15 commented on clusters/hosted_control_planes/managing_hosted_bm_ibmz.adoc:
```diff
@@ -0,0 +1,16 @@
+You must configure hosted control planes for bare metal before you can create hosted control plane clusters. See xref:..//hosted_control_planes/configure_hosted_bm_ibmz.adoc#configuring-hosting-service-cluster-configure-ibmz[Configuring the hosting cluster on 'x86' bare metal for IBM Z compute nodes (Technology Preview)] for more information.
+
+
+[#additional-resources-manage-bm]
```
```suggestion
[#additional-resources-manage-ibmz]
```

krishnanshruthi replied:
a. I did not change this -it is what it was or has been from the beginning.
b. The additional resources is common and the link given in the content for additional resources is not specific to ibm Z. So why are we changing the ID.

@SNiemann15

SNiemann15 commented on clusters/hosted_control_planes/configure_hosted_bm_ibmz.adoc:
```diff
@@ -0,0 +1,69 @@
+
+- When you scale down a node pool, Agents are unbound from the corresponding cluster. Before you reuse the clusters, you must boot the clusters by using the PXE image to update the number of nodes.
+
+[#hosting-service-cluster-configure-prereq]
```
```suggestion
[#hosting-service-cluster-configure-prereq-ibmz]
```

krishnanshruthi replied:
Same comment as before - Additionally i noticed that the rest of the sub-headings, dont necessarily have IBMz suffixed to the titles.

@SNiemann15

swopebe replied:
Please use the IBM z so that our team when creating links and such can better see where this file belongs. It is not as easy here as it was at IBM where the filenames are in the URLs. Thanks.

krishnanshruthi committed:
formatted the content as discussed with Silke

SNiemann15 commented on clusters/hosted_control_planes/config_dns_bare-metal_ibmz.adoc:
```diff
@@ -0,0 +1,37 @@
+
+[source,bash]
+----
+# cat /var/named/example.krnl.es.zone 
+$TTL 900
+@                     IN SOA bastion.example.krnl.es.com. hostmaster.example.krnl.es.com. (
+                        2019062002 1D 1H 1W 3H
+                      )
+                      IN NS bastion.example.krnl.es.com.
+api                   IN A 1xx.2x.2xx.1xx
+api-int               IN A 1xx.2x.2xx.1xx
+----
+----
+*`api.${HOSTED_CLUSTER_NAME}.${BASEDOMAIN}`                IN A 1xx.2x.2xx.1xx
+----
+
+*Note:* In the previous example, `1xx.2x.2xx.1xx` is a load balancer IP address that handles ingress and egress traffic for hosted control plane.
```
Not sure if I'm correct but I think ..cat is the command and starting with  $TTL is the output. 
Tagging [source,text] should resolve the * issue 
```suggestion
[source,terminal]
----
$ cat /var/named/<example.krnl.es.zone>  
----
+
.Example Output 
[source,text]
----
$ TTL 900
@ IN  SOA bastion.example.krnl.es.com. hostmaster.example.krnl.es.com. (
      2019062002
      1D 1H 1W 3H )
  IN NS bastion.example.krnl.es.com.
;
;
api                   IN A 1xx.2x.2xx.1xx <1>
api-int               IN A 1xx.2x.2xx.1xx
;
;
*api.${HOSTED_CLUSTER_NAME}.${BASEDOMAIN}           IN A 1xx.2x.2xx.1xx
;
;EOF
----
<1> The record refers to the IP address of the API load balancer that handles ingress and egress traffic for hosted control planes.
```

swopebe replied:
Hi! 

The bold on line 33 needs to be removed, being caused by the asterisks. 

I am not completely convinced that we need an full _example output_ in the doc...I want to make sure we can pause and ask if this is necessary. 

We have to maintain these outputs...meaning if anything changes, we are out of date without someone creating an issue for us. It may be better to tell the user what to search for in an output, unless the step is: 

See the following example and ensure you have X, Y, Z. (This would indicate that they have to search for something from the example.)

Also please change to:

See the following example output: (line 19, as that is how we introduce an example).

We have a number of conflicts that I will have to try to address tomorrow or the next day...just noting.

krishnanshruthi committed:
Brandi's review comment

lahinson commented on clusters/hosted_control_planes/config_dns_bare-metal_ibmz.adoc:
```diff
@@ -0,0 +1,37 @@
+[#configuring-dns-hosted-control-plane-ibmz]
+= Configure DNS for hosted control plane
```
```suggestion
= Configuring DNS for hosted control planes on IBM Z
```

lahinson commented on clusters/hosted_control_planes/config_dns_bare-metal_ibmz.adoc:
```diff
@@ -0,0 +1,37 @@
+[#configuring-dns-hosted-control-plane-ibmz]
+= Configure DNS for hosted control plane
+
+The API Server for the hosted cluster is exposed as a 'NodePort' service. A DNS entry must exist for the `api.<hosted_cluster_name>.<base-domain>` that points to the destination where the API Server is reachable.
```
```suggestion
The API server for the hosted cluster is exposed as a 'NodePort' service. A DNS entry must exist for the `api.<hosted_cluster_name>.<base-domain>` that points to the destination where the API server is reachable.
```
Unless there is a specific reason to capitalize "Server", the "s" in "API server" should be lowercase.

lahinson commented on clusters/hosted_control_planes/config_dns_bare-metal_ibmz.adoc:
```diff
@@ -0,0 +1,37 @@
+
+*Note:* In the previous example, `1xx.2x.2xx.1xx` is a load balancer IP address that handles ingress and egress traffic for hosted control plane.
+
+For IBM z/VM, add additional IP addresses that correspond to the IP address of the agent.
```
```suggestion
For IBM z/VM, add IP addresses that correspond to the IP address of the agent.
```
The phrase "add additional" can be simplified to "add".

lahinson commented on clusters/hosted_control_planes/configure_hosted_bm_ibmz.adoc:
```diff
@@ -0,0 +1,68 @@
+[#configuring-hosting-service-cluster-ibmz]
+= Configuring the hosting cluster on `x86` bare metal for IBM Z compute nodes (Technology Preview)
+
+**Technology Preview:** Configuring the hosting cluster on 'x86' bare metal is in a Technology Preview status with limited support.
```
On line 4, how are we intending to format "x86"? Should it be in monospace, as it is in the heading? If so, replace the single quotation marks with back ticks.

lahinson commented on clusters/hosted_control_planes/configure_hosted_bm_ibmz.adoc:
```diff
@@ -0,0 +1,68 @@
+
+The {mce-short} 2.4 supports only the default `local-cluster`, which is a hub cluster that is managed, and the hub cluster as the hosting cluster.
+
+*Note:* IBM Z does not support ISO boot.
```
Instead of placing two admonitions side by side, I suggest moving "IBM Z does not support ISO boot." to be the first bullet in the "Important" list.

krishnanshruthi replied:
Done

lahinson commented on clusters/hosted_control_planes/configure_hosted_bm_ibmz.adoc:
```diff
@@ -0,0 +1,68 @@
+
+*Note:* The _management_ cluster is not the _managed_ cluster. A managed cluster is a cluster that the hub cluster manages.
+
+You can convert a managed cluster to a hosting cluster by using the `hypershift` add-on to deploy the HyperShift Operator on that cluster. Then, you can start to create the hosted cluster. 
```
Is the statement on line 10 still true? I thought that since the HyperShift feature is now enabled by default, you don't have to explicitly do anything with the add-on anymore.

lahinson commented on clusters/hosted_control_planes/configure_hosted_bm_ibmz.adoc:
```diff
@@ -0,0 +1,68 @@
+
+**Important:**
+
+- To provision hosted control planes on bare metal, you can use the Agent platform. The Agent platform uses the Central Infrastructure Management (CIM) service to add worker nodes to a hosted cluster. For an introduction to the CIM service, see link:https://github.com/openshift/assisted-service/blob/master/docs/hive-integration/kube-api-getting-started.md[Kube API - Getting Started Guide].
```
```suggestion
- To provision hosted control planes on bare metal, you can use the Agent platform. The Agent platform uses the central infrastructure management service to add worker nodes to a hosted cluster. For an introduction to the central infrastructure management service, see link:https://github.com/openshift/assisted-service/blob/master/docs/hive-integration/kube-api-getting-started.md[Kube API - Getting Started Guide].
```

krishnanshruthi replied:
Done

lahinson commented on clusters/hosted_control_planes/configure_hosted_bm_ibmz.adoc:
```diff
@@ -0,0 +1,68 @@
+
+- To provision hosted control planes on bare metal, you can use the Agent platform. The Agent platform uses the Central Infrastructure Management (CIM) service to add worker nodes to a hosted cluster. For an introduction to the CIM service, see link:https://github.com/openshift/assisted-service/blob/master/docs/hive-integration/kube-api-getting-started.md[Kube API - Getting Started Guide].
+
+- Each IBM Z system host must be started with the PXE images provided by the CIM. After each host starts, it runs an Agent process to discover the details of the host and completes the installation. An Agent custom resource represents each host.
```
```suggestion
- Each IBM Z system host must be started with the PXE images provided by the central infrastructure management service. After each host starts, it runs an Agent process to discover the details of the host and completes the installation. An Agent custom resource represents each host.
```

krishnanshruthi replied:
Done

lahinson commented on clusters/hosted_control_planes/configure_hosted_bm_ibmz.adoc:
```diff
@@ -0,0 +1,68 @@
+
+- Each IBM Z system host must be started with the PXE images provided by the CIM. After each host starts, it runs an Agent process to discover the details of the host and completes the installation. An Agent custom resource represents each host.
+
+- When you create a hosted cluster with the Agent platform, HyperShift installs the Agent Cluster API provider in the hosted control plane namespace.
```
On line 22, what do you mean by "HyperShift"? Are you referring to the HyperShift Operator? If so, please use "HyperShift Operator". If you're referring to the hosted control planes feature, use "hosted control planes".

lahinson commented on clusters/hosted_control_planes/configure_hosted_bm_ibmz.adoc:
```diff
@@ -0,0 +1,68 @@
+* You need the {mce-short} must have at least one managed {ocp-short} cluster. The `local-cluster` is automatically imported in {mce-short} 2.4 and later. See xref:../install_upgrade/adv_config_install.adoc#advanced-config-engine[Advanced configuration] for more information about the `local-cluster`. You can check the status of your hub cluster by running the following command:
+
++
+[source,terminal]
```
```suggestion
[source,bash]
```
I believe the RHACM docs team uses `[source,bash]` instead of `[source,terminal]`. The OCP docs use `[source,terminal]` for most commands and `[source,bash]` for bash scripts.

krishnanshruthi replied:
Noted

lahinson commented on clusters/hosted_control_planes/configure_hosted_bm_ibmz.adoc:
```diff
@@ -0,0 +1,68 @@
+
+* You need a hosting cluster with at least three worker nodes to run the HyperShift Operator.
+
+* You need to enable Central Infrastructure Management (CIM). For more information, see xref:../cluster_lifecycle/create_infra_env.adoc#enable-cim[Enabling the Central Infrastructure Management service].
```
```suggestion
* You need to enable the central infrastructure management service. For more information, see xref:../cluster_lifecycle/create_infra_env.adoc#enable-cim[Enabling the Central Infrastructure Management service].
```

krishnanshruthi replied:
Done

lahinson commented on clusters/hosted_control_planes/configure_hosted_bm_ibmz.adoc:
```diff
@@ -0,0 +1,68 @@
+* DNS: The API and Ingress endpoints must be routable.
+
+[#configuring-hcp-bare-metal-ibmz]
+== Configuring hosted control planes on 'x86' bare metal for IBM Z compute nodes
```
```suggestion
== Next steps
```
Per recent conversations with @swopebe, we decided to standardize on the "Next steps" heading in cases like this, where we're including links to next steps.

krishnanshruthi replied:
Done

lahinson commented on clusters/hosted_control_planes/hosted-bm-agents-ibmz.adoc:
```diff
@@ -0,0 +1,120 @@
+[#hosted-bare-metal-adding-agents-ibmz]
+= Adding IBM Z agents to the InfraEnv resource (Technology Preview)
+
+Boot KVM on IBM Z and z/VM with the downloaded PXE images from the 'InfraEnv' resource. After the Agents are created, the host communicates with the Assisted Service and registers in the same namespace as the 'InfraEnv' resource on the management cluster.
```
```suggestion
Start KVM on IBM Z and IBM z/VM with the downloaded PXE images from the `InfraEnv` resource. After the Agents are created, the host communicates with the Assisted Service and registers in the same namespace as the `InfraEnv` resource on the management cluster.
```
Do we need to spell out "KVM"? I'm not sure what it refers to, but perhaps it's a term that most IBM Z users are familiar with. Just wanted to check.

krishnanshruthi replied:
Checked with team -z/VM and KVM are pretty familiar terms apparently. So, dont need to expand

lahinson commented on clusters/hosted_control_planes/hosted-bm-agents-ibmz.adoc:
```diff
@@ -0,0 +1,120 @@
+
+- For KVM on IBM Z, perform the boot by running the following command:
++
+[source,terminal]
```
```suggestion
[source,bash]
```

lahinson commented on clusters/hosted_control_planes/hosted-bm-agents-ibmz.adoc:
```diff
@@ -0,0 +1,120 @@
+
+Boot KVM on IBM Z and z/VM with the downloaded PXE images from the 'InfraEnv' resource. After the Agents are created, the host communicates with the Assisted Service and registers in the same namespace as the 'InfraEnv' resource on the management cluster.
+
+- For KVM on IBM Z, perform the boot by running the following command:
```
```suggestion
- For KVM on IBM Z, run the following command:
```
IBM Style Word Usage lists both "perform" and "boot" as terms to avoid, if possible.

lahinson commented on clusters/hosted_control_planes/hosted-bm-agents-ibmz.adoc:
```diff
@@ -0,0 +1,120 @@
+   --wait=-1 \
+   --extra-args "rd.neednet=1 nameserver=<nameserver>   coreos.live.rootfs_url=http://<http_server>/rootfs.img random.trust_cpu=on rd.luks.options=discard ignition.firstboot ignition.platform.id=metal console=tty1 console=ttyS1,115200n8 coreos.inst.persistent-kargs=console=tty1 console=ttyS1,115200n8"
+----  
+- For IBM z/VM, perform the boot by using the following procedure:
```
```suggestion
- For IBM z/VM, start KVM by completing the following procedure:
```

krishnanshruthi replied:
Done

lahinson commented on clusters/hosted_control_planes/hosted-bm-agents-ibmz.adoc:
```diff
@@ -0,0 +1,120 @@
+----  
+- For IBM z/VM, perform the boot by using the following procedure:
+
+. Update the paramfile to add `rootfs url`, `net type` and `disk type`. 
```
```suggestion
. Update the paramfile to add the `rootfs url`, `net type` and `disk type` values. 
```

lahinson commented on clusters/hosted_control_planes/hosted-bm-agents-ibmz.adoc:
```diff
@@ -0,0 +1,120 @@
+
+. Update the paramfile to add `rootfs url`, `net type` and `disk type`. 
+
+Following is the sample command for 'VSwitch' network with fcp singlepath/ dasd disk:
```
```suggestion
+
The following example shows a  sample command for the `VSwitch` network with a `fcp singlepath/ dasd` disk:
```

lahinson commented on clusters/hosted_control_planes/hosted-bm-agents-ibmz.adoc:
```diff
@@ -0,0 +1,120 @@
+
+Following is the sample command for 'VSwitch' network with fcp singlepath/ dasd disk:
++
+[source,terminal]
```
```suggestion
[source,bash]
```

lahinson commented on clusters/hosted_control_planes/hosted-bm-agents-ibmz.adoc:
```diff
@@ -0,0 +1,120 @@
+rd.neednet=1 console=ttysclp0   coreos.live.rootfs_url=<rootfs_url> ip=<IP for guest vm>::<nameserver>:255.255.255.0::<network adaptor>:none nameserver=<nameserver> zfcp.allow_lun_scan=0  rd.znet=qeth,<network adaptor range>,layer2=1 rd.<zfcp/dasd>=<storage> random.trust_cpu=on rd.luks.options=discard ignition.firstboot ignition.platform.id=metal console=tty1 console=ttyS1,115200n8 coreos.inst.pe ...
+----  
+
+Following is the sample command for 'OSA' & 'Hipersocket' network with fcp singlepath/dasd disk:
```
```suggestion
+
The following example is a sample command for the `OSA` and `Hipersocket` networks with a `fcp singlepath/dasd` disk:
```

lahinson commented on clusters/hosted_control_planes/hosted-bm-agents-ibmz.adoc:
```diff
@@ -0,0 +1,120 @@
+
+Following is the sample command for 'OSA' & 'Hipersocket' network with fcp singlepath/dasd disk:
++
+[source,terminal]
```
```suggestion
[source,bash]
```

lahinson commented on clusters/hosted_control_planes/hosted-bm-agents-ibmz.adoc:
```diff
@@ -0,0 +1,120 @@
+rd.neednet=1 console=ttysclp0   coreos.live.rootfs_url=<rootfs_url> ip=<IP for guest vm>::<nameserver>:255.255.255.0::<network adaptor>:none nameserver=<nameserver> rd.znet=qeth,<network adaptor range>,layer2=1 rd.<zfcp/dasd>=<storage> random.trust_cpu=on rd.luks.options=discard ignition.firstboot ignition.platform.id=metal console=tty1 console=ttyS1,115200n8 coreos.inst.persistent-kargs="console ...
+----
+
+Following is the sample command for 'ROCE' network with fcp singlepath/ dasdc disk:
```
```suggestion
+
The following example is a sample command for the `ROCE` network with a `fcp singlepath/ dasdc` disk:
```

lahinson commented on clusters/hosted_control_planes/hosted-bm-agents-ibmz.adoc:
```diff
@@ -0,0 +1,120 @@
+
+Following is the sample command for 'ROCE' network with fcp singlepath/ dasdc disk:
++
+[source,terminal]
```
```suggestion
[source,bash]
```

lahinson commented on clusters/hosted_control_planes/hosted-bm-agents-ibmz.adoc:
```diff
@@ -0,0 +1,120 @@
+rd.neednet=1 console=ttysclp0         coreos.live.rootfs_url=<rootfs_url> ip=<IP for guest vm>::<nameserver>:255.255.255.0 nameserver=<nameserver> rd.<zfcp/dasd>=<storage> random.trust_cpu=on rd.luks.options=discard ignition.firstboot ignition.platform.id=metal console=tty1 console=ttyS1,115200n8 coreos.inst.persistent-kargs="console=tty1 console=ttyS1,115200n8" 
+----  
+
+. Punch 'initrd',  kernel images and paramfile to guest VM by running following commands:
```
```suggestion
. Punch `initrd`,  kernel images, and the paramfile to the guest VM by running the following commands:
```
IBM Style warns against using terms that might be associated with violence, such as "punch". Can we change this line to something like this? "Move `initrd`, kernel images, and the paramfile to the guest VM by running the following commands:"

lahinson commented on clusters/hosted_control_planes/hosted-bm-agents-ibmz.adoc:
```diff
@@ -0,0 +1,120 @@
+. Punch 'initrd',  kernel images and paramfile to guest VM by running following commands:
+
++
+[source,terminal]
```
```suggestion
[source,bash]
```

lahinson commented on clusters/hosted_control_planes/hosted-bm-agents-ibmz.adoc:
```diff
@@ -0,0 +1,120 @@
+vmur pun -r -u -N kernel.img $INSTALLERKERNELLOCATION/<image name>
+vmur pun -r -u -N generic.parm $PARMFILELOCATION/paramfilename
+vmur pun -r -u -N initrd.img $INSTALLERINITRAMFSLOCATION/<image name>
+
```
Is line 57 left blank intentionally?

lahinson commented on clusters/hosted_control_planes/hosted-bm-agents-ibmz.adoc:
```diff
@@ -0,0 +1,120 @@
+vmur pun -r -u -N initrd.img $INSTALLERINITRAMFSLOCATION/<image name>
+
+----  
+. Run the following command from the guest VM console to perform the boot.
```
```suggestion
. Run the following command from the guest VM console to start KVM:
```

lahinson commented on clusters/hosted_control_planes/hosted-bm-agents-ibmz.adoc:
```diff
@@ -0,0 +1,120 @@
+----  
+. Run the following command from the guest VM console to perform the boot.
++
+[source,terminal]
```
```suggestion
[source,bash]
```

lahinson commented on clusters/hosted_control_planes/hosted-bm-agents-ibmz.adoc:
```diff
@@ -0,0 +1,120 @@
+----
+
+- To list the agents and their properties, enter the following command:
++
```
Add a blank line above the + to ensure proper formatting.

lahinson commented on clusters/hosted_control_planes/hosted-bm-agents-ibmz.adoc:
```diff
@@ -0,0 +1,120 @@
+
+- To list the agents and their properties, enter the following command:
++
+[source,terminal]
```
```suggestion
[source,bash]
```

lahinson commented on clusters/hosted_control_planes/hosted-bm-agents-ibmz.adoc:
```diff
@@ -0,0 +1,120 @@
++
+[source,terminal]
+----
+oc -n ${HOSTED_CONTROL_PLANE_NAMESPACE} get agents
```
```suggestion
oc -n <hosted_control_plane_namespace> get agents
```

lahinson commented on clusters/hosted_control_planes/hosted-bm-agents-ibmz.adoc:
```diff
@@ -0,0 +1,120 @@
+----
+oc -n ${HOSTED_CONTROL_PLANE_NAMESPACE} get agents
+----
+.Example output
```
Add a blank line between the `----` on line 71 and `.Example output` on line 72 to ensure proper formatting.

lahinson commented on clusters/hosted_control_planes/hosted-bm-agents-ibmz.adoc:
```diff
@@ -0,0 +1,120 @@
+----
+.Example output
++
+[source,terminal]
```
```suggestion
[source,bash]
```

lahinson commented on clusters/hosted_control_planes/managing_hosted_kubevirt.adoc:
I'm not sure why this file is included in this PR. I suspect it might have been included accidentally, as `configure_hosted_aws.adoc` and `configure.hosted.bm.adoc` were. I think we should remove this from the PR.

lahinson commented on clusters/hosted_control_planes/hosted-bm-agents-ibmz.adoc:
```diff
@@ -0,0 +1,120 @@
+NAME    CLUSTER APPROVED    ROLE    STAGE
+50c23cda-cedc-9bbd-bcf1-9b3a5c75804d    auto-assign
+5e498cd3-542c-e54f-0c58-ed43e28b568a    auto-assign
+
```
Is line 79 left blank intentionally?

lahinson commented on clusters/hosted_control_planes/hosted-bm-agents-ibmz.adoc:
```diff
@@ -0,0 +1,120 @@
+5e498cd3-542c-e54f-0c58-ed43e28b568a    auto-assign
+
+----
+
```
Something strange is happening with the formatting starting around line 82. I'm happy to troubleshoot it after we can get some of the other comments resolved, as it will be easier for me to investigate then.

lahinson commented on clusters/hosted_control_planes/hosted-bm-agents-ibmz.adoc:
```diff
@@ -0,0 +1,120 @@
+Optionally, you can set the agent ID `installation_disk_id` and hostname in the specification. 
+
+. Run the following command to approve the agent:
++
```
Add a blank line between the step on line 84 and the + on line 85.

lahinson commented on clusters/hosted_control_planes/hosted-bm-agents-ibmz.adoc:
```diff
@@ -0,0 +1,120 @@
+
+. Run the following command to approve the agent:
++
+[source,terminal]
```
```suggestion
[source,bash]
```

lahinson commented on clusters/hosted_control_planes/hosted-bm-agents-ibmz.adoc:
```diff
@@ -0,0 +1,120 @@
++
+[source,terminal]
+----
+oc -n ${HOSTED_CONTROL_PLANE_NAMESPACE} patch agent 50c23cda-cedc-9bbd-bcf1-9b3a5c75804d -p '{"spec":{"installation_disk_id":"/dev/sda","approved":true,"hostname":"worker-zvm-0.hostedn.example.com"}}' --type merge
```
```suggestion
oc -n <hosted_control_plane_namespace> patch agent 50c23cda-cedc-9bbd-bcf1-9b3a5c75804d -p '{"spec":{"installation_disk_id":"/dev/sda","approved":true,"hostname":"worker-zvm-0.hostedn.example.com"}}' --type merge
```

lahinson commented on clusters/hosted_control_planes/hosted-bm-agents-ibmz.adoc:
```diff
@@ -0,0 +1,120 @@
+
+. Run the following command to verify that the agents are approved:
++
+[source,terminal]
```
```suggestion
[source,bash]
```

lahinson commented on clusters/hosted_control_planes/hosted-bm-agents-ibmz.adoc:
```diff
@@ -0,0 +1,120 @@
+----
+
+. Run the following command to verify that the agents are approved:
++
```
Add a blank line between lines 91 and 92 to ensure proper formatting.

lahinson commented on clusters/hosted_control_planes/hosted-bm-agents-ibmz.adoc:
```diff
@@ -0,0 +1,120 @@
++
+[source,terminal]
+----
+oc -n ${HOSTED_CONTROL_PLANE_NAMESPACE} get agents
```
```suggestion
oc -n <hosted_control_plane_namespace> get agents
```

lahinson commented on clusters/hosted_control_planes/hosted-bm-agents-ibmz.adoc:
```diff
@@ -0,0 +1,120 @@
+----
+oc -n ${HOSTED_CONTROL_PLANE_NAMESPACE} get agents
+----
+.Example output
```
Add a blank line between lines 96 and 97 to ensure proper formatting.

lahinson commented on clusters/hosted_control_planes/hosted-bm-agents-ibmz.adoc:
```diff
@@ -0,0 +1,120 @@
+----
+.Example output
++
+[source,terminal]
```
```suggestion
[source,bash]
```

lahinson commented on clusters/hosted_control_planes/hosted-bm-agents-ibmz.adoc:
```diff
@@ -0,0 +1,120 @@
+5e498cd3-542c-e54f-0c58-ed43e28b568a             true       auto-assign
+----
+
+. For IBM z/VM, an additional step is required to edit the agent patched in the last step and update installer 'args' under 'spec' section as follows:
```
```suggestion
. For IBM z/VM, edit the agent that you patched in the previous step and update the `installerArgs` setting in the `spec` section as follows:
```

lahinson commented on clusters/hosted_control_planes/hosted-bm-agents-ibmz.adoc:
```diff
@@ -0,0 +1,120 @@
+
+. For IBM z/VM, an additional step is required to edit the agent patched in the last step and update installer 'args' under 'spec' section as follows:
+
+- For network types as VSwitch/ OSA & Hipersocket:
```
```suggestion
- For network types such as `VSwitch`, `OSA`, and `Hipersocket`, enter the following command:
```

lahinson commented on clusters/hosted_control_planes/hosted-bm-agents-ibmz.adoc:
```diff
@@ -0,0 +1,120 @@
+
+- For network types as VSwitch/ OSA & Hipersocket:
++
+[source,terminal]
```
```suggestion
[source,bash]
```

lahinson commented on clusters/hosted_control_planes/hosted-bm-agents-ibmz.adoc:
```diff
@@ -0,0 +1,120 @@
+installerArgs: '[ "--append-karg", "rd.neednet=1", "--append-karg", "ip=<Guest vm ip>::<nameserver>:255.255.255.0:<hostname>:<network adaptor>:none", "--append-karg", "nameserver=<nameserver>", "--append-karg", "rd.znet=qeth,<network adaptor range>,layer2=1", "--append-karg", "rd.<storage type>=<storage>" ]'
+----
+
+- For network type as ROCE:
```
```suggestion
- For network types such as `ROCE`, enter the following command:
```

lahinson commented on clusters/hosted_control_planes/hosted-bm-agents-ibmz.adoc:
```diff
@@ -0,0 +1,120 @@
+
+- For network type as ROCE:
++
+[source,terminal]
```
```suggestion
[source,bash]
```

lahinson commented on clusters/hosted_control_planes/hosted_infraenv_ibmz.adoc:
```diff
@@ -0,0 +1,38 @@
+
++
+----
+oc -n ${HOSTED_CONTROL_PLANE_NAMESPACE} get InfraEnv ${HOSTED_CLUSTER_NAME} -o json
```
```suggestion
oc -n <hosted_control_plane_namespace> get InfraEnv <hosted_cluster_name> -o json
```

krishnanshruthi replied:
Done

lahinson commented on clusters/hosted_control_planes/scaling_nodepool_hosted_ibmz.adoc:
```diff
@@ -0,0 +1,137 @@
+[#scaling-the-nodepool-ibmz]
+= Scaling the NodePool object for a hosted cluster on IBM Z
+
+The NodePool object is created when creating a hosted cluster. By scaling the NodePool object, you can add more compute nodes to the Hosted Control Plane.
```
```suggestion
The `NodePool` object is created when you create a hosted cluster. By scaling the `NodePool` object, you can add more compute nodes to the hosted control plane.
```

lahinson commented on clusters/hosted_control_planes/scaling_nodepool_hosted_ibmz.adoc:
```diff
@@ -0,0 +1,137 @@
+
+The NodePool object is created when creating a hosted cluster. By scaling the NodePool object, you can add more compute nodes to the Hosted Control Plane.
+
+. Run the following command to scale the NodePool object to two nodes:
```
```suggestion
. Run the following command to scale the `NodePool` object to two nodes:
```

lahinson commented on clusters/hosted_control_planes/scaling_nodepool_hosted_ibmz.adoc:
```diff
@@ -0,0 +1,137 @@
+. Run the following command to scale the NodePool object to two nodes:
+
++
+[source,terminal]
```
```suggestion
[source,bash]
```

lahinson commented on clusters/hosted_control_planes/scaling_nodepool_hosted_ibmz.adoc:
```diff
@@ -0,0 +1,137 @@
++
+[source,terminal]
+----
+oc -n $<CLUSTERS_NAMESPACE> scale nodepool $<NODEPOOL_NAME> --replicas 2
```
```suggestion
oc -n <clusters_namespace> scale nodepool <nodepool_name> --replicas 2
```

lahinson commented on clusters/hosted_control_planes/scaling_nodepool_hosted_ibmz.adoc:
```diff
@@ -0,0 +1,137 @@
+oc -n $<CLUSTERS_NAMESPACE> scale nodepool $<NODEPOOL_NAME> --replicas 2
+----
++
+The Cluster API agent provider randomly picks two agents that are then assigned to the hosted cluster. Those agents go through different states and finally join the hosted cluster as {ocp-short} nodes. The agents pass through the transition phases as given in the following order:
```
```suggestion
The Cluster API agent provider randomly picks two agents that are then assigned to the hosted cluster. Those agents go through different states and finally join the hosted cluster as {ocp-short} nodes. The agents pass through the transition phases in the following order:
```

lahinson commented on clusters/hosted_control_planes/scaling_nodepool_hosted_ibmz.adoc:
```diff
@@ -0,0 +1,137 @@
+* `added-to-existing-cluster`
+
+. Run the following command to see the status of a specific scaled agent:
++
```
Add a blank line between lines 23 and 24 to ensure proper formatting.

lahinson commented on clusters/hosted_control_planes/scaling_nodepool_hosted_ibmz.adoc:
```diff
@@ -0,0 +1,137 @@
+
+. Run the following command to see the status of a specific scaled agent:
++
+[source,terminal]
```
```suggestion
[source,bash]
```

lahinson commented on clusters/hosted_control_planes/scaling_nodepool_hosted_ibmz.adoc:
```diff
@@ -0,0 +1,137 @@
++
+[source,terminal]
+----
+c -n ${HOSTED_CONTROL_PLANE_NAMESPACE} get agent -o jsonpath='{range .items[*]}BMH: {@.metadata.labels.agent-install\.openshift\.io/bmh} Agent: {@.metadata.name} State: {@.status.debugInfo.state}{"\n"}{end}'
```
```suggestion
oc -n <hosted_control_plane_namespace> get agent -o jsonpath='{range .items[*]}BMH: {@.metadata.labels.agent-install\.openshift\.io/bmh} Agent: {@.metadata.name} State: {@.status.debugInfo.state}{"\n"}{end}'
```

lahinson commented on clusters/hosted_control_planes/scaling_nodepool_hosted_ibmz.adoc:
```diff
@@ -0,0 +1,137 @@
++
+.Example output
++
+[source,terminal]
```
```suggestion
[source,bash]
```

lahinson commented on clusters/hosted_control_planes/scaling_nodepool_hosted_ibmz.adoc:
```diff
@@ -0,0 +1,137 @@
+. Run the following command to see the transition phases:
+
++
+[source,terminal]
```
```suggestion
[source,bash]
```

lahinson commented on clusters/hosted_control_planes/scaling_nodepool_hosted_ibmz.adoc:
```diff
@@ -0,0 +1,137 @@
++
+[source,terminal]
+----
+oc -n ${HOSTED_CONTROL_PLANE_NAMESPACE} get agent
```
```suggestion
oc -n <hosted_control_plane_namespace> get agent
```

lahinson commented on clusters/hosted_control_planes/scaling_nodepool_hosted_ibmz.adoc:
```diff
@@ -0,0 +1,137 @@
++
+.Example output
++
+[source,terminal]
```
```suggestion
[source,bash]
```

lahinson commented on clusters/hosted_control_planes/scaling_nodepool_hosted_ibmz.adoc:
```diff
@@ -0,0 +1,137 @@
+da503cf1-a347-44f2-875c-4960ddb04091   hosted-forwarder   true          auto-assign
+----
+
+. Run the following command to generate the kubeconfig to access the hosted cluster.
```
```suggestion
. Run the following command to generate the `kubeconfig` file to access the hosted cluster:
```

lahinson commented on clusters/hosted_control_planes/scaling_nodepool_hosted_ibmz.adoc:
```diff
@@ -0,0 +1,137 @@
+
+. Run the following command to generate the kubeconfig to access the hosted cluster.
++
+[source,terminal]
```
```suggestion
[source,bash]
```

lahinson commented on clusters/hosted_control_planes/scaling_nodepool_hosted_ibmz.adoc:
```diff
@@ -0,0 +1,137 @@
++
+[source,terminal]
+----
+hypershift create kubeconfig --namespace ${CLUSTERS_NAMESPACE} --name ${HOSTED_CLUSTER_NAME} > ${HOSTED_CLUSTER_NAME}.kubeconfig
```
```suggestion
hypershift create kubeconfig --namespace <clusters_namespace> --name <hosted_cluster_namespace> > <hosted_cluster_name>.kubeconfig
```
Please check with your engineering contact to confirm that `hypershift create` is the correct command. In the bare metal and kubevirt docs, we no longer refer to the `hypershift` CLI commands and instead refer to the `hcp` CLI commands.

lahinson commented on clusters/hosted_control_planes/scaling_nodepool_hosted_ibmz.adoc:
```diff
@@ -0,0 +1,137 @@
+
+. After the agents reach the `added-to-existing-cluster` state, verify that you can see the {ocp-short} nodes by entering the following command:
++
+[source,terminal]
```
```suggestion
[source,bash]
```

lahinson commented on clusters/hosted_control_planes/scaling_nodepool_hosted_ibmz.adoc:
```diff
@@ -0,0 +1,137 @@
++
+[source,terminal]
+----
+oc --kubeconfig ${HOSTED_CLUSTER_NAME}.kubeconfig get nodes
```
```suggestion
oc --kubeconfig <hosted_cluster_name>.kubeconfig get nodes
```

lahinson commented on clusters/hosted_control_planes/scaling_nodepool_hosted_ibmz.adoc:
```diff
@@ -0,0 +1,137 @@
++
+.Example output
++
+[source,terminal]
```
```suggestion
[source,bash]
```

lahinson commented on clusters/hosted_control_planes/scaling_nodepool_hosted_ibmz.adoc:
```diff
@@ -0,0 +1,137 @@
+
+. Enter the following command to verify that two machines were created when you scaled up the `NodePool` object:
++
+[source,terminal]
```
```suggestion
[source,bash]
```

lahinson commented on clusters/hosted_control_planes/scaling_nodepool_hosted_ibmz.adoc:
```diff
@@ -0,0 +1,137 @@
++
+[source,terminal]
+----
+oc -n ${HOSTED_CONTROL_PLANE_NAMESPACE} get machine.cluster.x-k8s.io
```
```suggestion
oc -n <hosted_control_plane_namespace> get machine.cluster.x-k8s.io
```

lahinson commented on clusters/hosted_control_planes/scaling_nodepool_hosted_ibmz.adoc:
```diff
@@ -0,0 +1,137 @@
++
+.Example output
++
+[source,terminal]
```
```suggestion
[source,bash]
```

lahinson commented on clusters/hosted_control_planes/scaling_nodepool_hosted_ibmz.adoc:
```diff
@@ -0,0 +1,137 @@
+. Run the following command to check the `clusterversion` and cluster operator status:
+
++
+[source,terminal]
```
```suggestion
[source,bash]
```

lahinson commented on clusters/hosted_control_planes/scaling_nodepool_hosted_ibmz.adoc:
```diff
@@ -0,0 +1,137 @@
++
+[source,terminal]
+----
+oc --kubeconfig ${HOSTED_CLUSTER_NAME}.kubeconfig get clusterversion,co
```
```suggestion
oc --kubeconfig <hosted_cluster_name>.kubeconfig get clusterversion,co
```

lahinson commented on clusters/hosted_control_planes/scaling_nodepool_hosted_ibmz.adoc:
```diff
@@ -0,0 +1,137 @@
++
+.Example output
++
+[source,terminal]
```
```suggestion
[source,bash]
```

lahinson reviewed (changes requested):
@krishnanshruthi @SNiemann15 I provided several peer review comments along with @swopebe's comments. Let me know if you have any questions for me. Thanks!

krishnanshruthi committed:
updated as per Laura's review comments

krishnanshruthi committed:
updated as per silke's review comments

krishnanshruthi commented:
Follow the new PR https://github.com/stolostron/rhacm-docs/pull/5792 for further updates. Closing this PR.
cc: @SNiemann15

krishnanshruthi commented:
Follow the new PR https://github.com/stolostron/rhacm-docs/pull/5792 for further updates. Closing this PR.
cc: @SNiemann15
